app/src/lib/stores/app-store.ts

```diff
@@ -3019,11 +3039,14 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return repository
     }
 
+    const branches = await api.fetchProtectedBranches(owner, name)
```

Comment:
Could we feature flag this as well and use an empty array when the feature flag isn't turned on?